java/compaction/compaction-core/src/test/java/sleeper/compaction/core/job/dispatch/CompactionJobDispatcherTest.java

```diff
@@ -324,9 +339,40 @@ private CompactionJobDispatchRequest generateBatchRequestAtTime(String batchId,
                 tableProperties, batchId, timeNow);
     }
 
+    private CompactionJobDispatchRequest generateBatchRequest(String batchId) {
+        return CompactionJobDispatchRequest.forTableWithBatchIdAtTime(
+                tableProperties, batchId, Instant.now());
+    }
+
     private record BatchRequestMessage(CompactionJobDispatchRequest request, int delaySeconds) {
         static BatchRequestMessage requestAndDelay(CompactionJobDispatchRequest request, Duration duration) {
             return new BatchRequestMessage(request, (int) duration.toSeconds());
         }
     }
+
+    private void assertThatBatchIsQueuedForResend() {
+        assertThat(delayedPendingQueue).isEmpty();
+        assertThat(pendingDeadLetterQueue).isEmpty();
+        assertThat(s3PathToCompactionJobBatch).isNotEmpty();
+    }
+
+    private void assertThatNoBatchesAreQueuedForResend() {
+        assertThat(delayedPendingQueue).isEmpty();
+        assertThat(pendingDeadLetterQueue).isEmpty();
+        assertThat(s3PathToCompactionJobBatch).isEmpty();
+    }
+
+    private void assertThatBatchIsPendingResend(CompactionJobDispatchRequest request) {
+        assertThat(delayedPendingQueue).containsExactly(
+                BatchRequestMessage.requestAndDelay(request, Duration.ofSeconds(12)));
```

Comment:
It might be worth passing the delay in as a parameter, particularly as it's set explicitly in a table property in the tests that use this.